### ISSUE TITLE
Collect errors from automatic suggestion backends

### ISSUE THREAD

**Issue by nijel** [WeblateOrg/weblate#9638]
### Describe the problem

Currently, automatic suggestion backends silent themselves for errors which are most likely caused by rate limiting:

https://github.com/WeblateOrg/weblate/blob/8b514cb50db9a4839c0e48350d4d5c5c3fe13250/weblate/machinery/base.py#L246-L259

This is nice for user viewed suggestions as it doesn't show an error message on each page.

But this makes such errors invisible to admins.

### Describe the solution you'd like

Collect such errors and display them in the management interface.

**Comment by AbdullahHabib-github** [comment]
I've over 3 years of experience in Python development and would love to contribute. @nijel Can you please assign this issue to me?

**Comment by nijel** [comment]
@AbdullahHabib-github you're welcome!

**Comment by AbdullahHabib-github** [comment]
@nijel should I create a new view or am I supposed to integrate it in any of the existing views?

**Comment by nishitiwari22** [comment]
Is this issue still open? I am looking to contribute to it.

**Comment by nijel** [comment]
Yes, this is open. There is a similar change happening for add-ons, it should at least share the UI.

**Comment by nishitiwari22** [comment]
Okay, so I am trying to understand the particular Python file and I came up with a temporary code: 

To resolve the issue of rate-limiting errors being silenced for admins, I made changes in the is_rate_limit_error method of the MachineTranslation class in the weblate/weblate/machinery/base.py file.

 def is_rate_limit_error(self, exc):
        if isinstance(exc, MachineryRateLimit):
            return True
        if not isinstance(exc, HTTPError):
            return False
    # Apply rate limiting for following status codes:
    # HTTP 456 Client Error: Quota Exceeded (DeepL)
    # HTTP 429 Too Many Requests
    # HTTP 401 Unauthorized
    # HTTP 403 Forbidden
    # HTTP 503 Service Unavailable
        if exc.response.status_code in (456, 429, 401, 403, 503):
            return True
        return False

I am trying to locally run on my machine as well. I'll add more as a move forward if the maintainer allows. Actually, this is my first PR and I want to take some time making it a valuable one. Therefore before committing any change I thought to let the maintainer know about it.

**Comment by nijel** [comment]
The implementation should be:

* Add model to store the errors to `weblate/machinery/models.py`, it needs include engine name, nullable foreign key to `Project` (for project scoped installations), timestamp and field to store the error
* Extend `MachineTranslation.report_error` method to take exception paramater and save the error to the database:
  
  https://github.com/WeblateOrg/weblate/blob/8b514cb50db9a4839c0e48350d4d5c5c3fe13250/weblate/machinery/base.py#L203-L206
* Add UI to list these errors
* Add a background task to prune these errors after some time (30 days?)

**Comment by hemaa0325** [comment]
Hi, I’d like to work on this issue.

I’ll look into collecting rate-limit/backend errors from automatic suggestion backends and exposing them in the management/admin interface.

Please let me know if there is a preferred place/model where these machinery backend errors should be stored.

**Comment by nijel** [comment]
There is no existing model, but there is similar model for add-ons that might share some logic.

**Comment by shwetasinghh** [comment]
@nijel could you please review my PR here https://github.com/WeblateOrg/weblate/pull/20717
